### PR TITLE
Upload of AppMap files

### PR DESCRIPTION
**Pull request by jansorg**
Closes https://github.com/applandinc/appmap-intellij-plugin/issues/31

To be discussed:
- [x] The uploader doesn't enforce HTTPs. Please comment if a secure connection should be enforced.
- [x] If you want to keep it, could you provide an icon, please? For now, I used the file type's icon as fallback
- [x] In addition to the AppMap app's button, I included a separate action at `Tools>AppMap>Upload Current AppMap`. This is more IntelliJ-like, is discoverable with "Search action" and also allows to attach a shortcut. Do you want to keep it?
- [x] Should the server be configurable via settings, as in the VSCode extension?
- [x] Similar to VSCode, a request header is used: `X-Requested-With: IntelliJUploader`. Is this the correct value?
- [x] As in VSCode, the confirmation to upload is only needed for the first upload. It's stored in the project settings for all later uploads. Do you want to keep this? 
- [x] The confirmation setting isn't configurable. Should I add UI for the settings?

To do:
- [x] The upload endpoint isn't working for me yet. Testing is needed to finish this PR.
- [x] Add a unit test with the embedded HTTP server to verify the upload process.

## Settings UI
Screenshot_059

### CODE REVIEW

**Comment by ptrdvrk** [comment]
Hello @jansorg , here are answers for your questions:

1) Yes, let's keep the action. 
2) We'll have an updated upload icon soon.
3) Preferably yes, if it isn't a major complication. 
4) Yes, `IntelliJUploader` works.
5) Yes, let's keep the confirmation in project settings
6) Yes, that is a good plan to UI for the Settings

**Comment by jansorg** [comment]
@ptrdvrk This PR is ready as soon as the icon is available.

**Comment by ptrdvrk** [comment]
The HTTPS enforcement is not needed in the client configuration.

**Review by ptrdvrk — APPROVED** [review]
Looks OK, works well, approving.

File: src/test/java/appland/upload/AppMapUploaderTest.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+            var ok = latch.await(1000, TimeUnit.MILLISECONDS);
+            assertTrue("Upload response must not timeout", ok);
+
+            var expectedURL = String.format("%s/scenario_uploads/%d?token=%s",
+                    serverRule.baseUrl(), 248, "d70c362d-7bd3-4363-bb19-f8d88f6a3496");
+            assertEquals(expectedURL, result.get());
+        } finally {
+            projectSettings.setCloudServerUrl(null);
+            projectSettings.setConfirmAppMapUpload(null);
```
[the remark this answers is not in the document]
Comment (jansorg) [reply src/test/java/appland/upload/AppMapUploaderTest.java]: Fixed by now

**Pull request merged**